src/Core/Types/Configuration/TypeInitializer.cs

```diff
@@ -196,6 +196,14 @@ private void MergeTypeExtensions(DiscoveredTypes discoveredTypes)
                         && extension.Type is INamedTypeExtensionMerger m
                         && type.Type is INamedType n)
                     {
+                        // replace resolvers
```

Comment:
Ah, right almost forgot about this... in this case we have to set the source type. But we have to ensure that the resolverType is also correct. So we might need some extra tests here.

Reply:
What should I test for?